src/Akka.Streams.Kafka/Extensions/CollectionExtensions.cs

```diff
@@ -0,0 +1,34 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Akka.Streams.Kafka.Extensions
+{
+    public static class CollectionExtensions
+    {
+        /// <summary>
+        /// Joins elements of the collection with given separator to single string
+        /// </summary>
+        public static string JoinToString<T>(this IEnumerable<T> collection, string separator)
+        {
+            return string.Join(separator, collection);
+        }
+
+        /// <summary>
+        /// Converts dictionary to a list of tuples
+        /// </summary>
+        public static IEnumerable<(TKey, TValue)> ToTuples<TKey, TValue>(this IEnumerable<KeyValuePair<TKey, TValue>> dictionary)
```

Comment:
For the benefit of other OSS contributors: `ValueTuple` is a struct so this isn't really an "allocation" worth fretting about.

Reply:
Sorry, did not understand what change to you propose. I used this extension to be able to write code like that:
```csharp
foreach (var (first, second) in dictionary.ToTuples()) { }
```
Because `KeyValuePair` does not have `Deconstruct` method to be converted to `ValueTuple `automatically, but in this project such conversion is quite handy.

Reply:
I wasn't proposing any changes - your code is fine as-is. Was just stating for other members of our OSS org that this doesn't represent any unnecessary allocations since `ValueTuple`s are on the stack. Unnecessary allocations are a type of memory issue we look out for especially when working with collections.